pkg/api/v1/state.go

```diff
@@ -235,3 +235,54 @@ func (s Server) ListRowsHTTP(w http.ResponseWriter, r *http.Request) {
 
 	s.writeJSON(w, resp)
 }
+
+// GetDashboardTab returns config of a given dashboard tab
+func (s *Server) GetDashboardTab(ctx context.Context, req *apipb.GetDashboardTabRequest) (*apipb.GetDashboardTabResponse, error) {
+
+	// this should be factored out of this function
+	c, err := s.getConfig(ctx, logrus.WithContext(ctx), req.GetScope())
+
+	if err != nil {
+		return nil, err
+	}
+	c.Mutex.RLock()
+	defer c.Mutex.RUnlock()
+
+	dashboardName, tabName, testGroupName, err := findDashboardTab(c, req.GetDashboard(), req.GetTab())
+	if err != nil {
+		return nil, err
+	}
+
+	grid, err := s.Grid(ctx, req.GetScope(), dashboardName, tabName, testGroupName)
+	if err != nil {
+		return nil, fmt.Errorf("dashboard {%q} or tab {%q} not found", req.GetDashboard(), req.GetTab())
+	}
+	if grid == nil {
+		return nil, errors.New("grid not found")
+	}
+
+	dashboardTabResponse := apipb.GetDashboardTabResponse{
+		Name:           dashboardName,
+		TestGroupName:  testGroupName,
+		CodeSearchPath: "lorem ipsum",
```

Comment:
@michelle192837 @sultan-duisenbay I am not sure how will I fetch this data. Can you help me with this?

Reply:
@ankur12-1610 - don't worry about the CodeSearchPath for now. I think, Sean mentioned in the issue that we would like to see ResultSource, PrimaryGrouping and ColumnHeader fields...

In addition, `s.Grid()` downloads and unmarshals the grid data, while we are interested in fetching the data from config proto - so no need in looking at the grid data.

Essentially, we want to fetch the most important configuration data about the tab. 
To start, you have to know that  configuration consists of dashboardGroup, Dashboard and Testgroup configurations. Tabs configurations are part of Dashboard configuration - and Tab configurations include the testGroupName (think of it as a foreign key for searching within Testgroup configurations). The fields of interest are stored within that particular Testgroup configuration

1 - Find the correct Dashboard config
2 - Find the correct Tab config and store the testgroupname
3 - Find the correct Testgroup config using the name found in step 2
4 - Fetch the required data and make the API return it

Hope this helps!

Reply:
Are you working on this @sultan-duisenbay?

Reply:
Hi @ankur12-1610 ! No, I got confused and closed this PR instead of another - sorry for confusion

Reply:
No problem :), I didn't get time this week, so couldn't do any progress :/